sys/targets/targets.go

```diff
@@ -619,18 +623,25 @@ func init() {
 	}
 	for _, target := range List[TestOS] {
 		if List[goos] != nil {
-			if host := List[goos][goarch]; host != nil {
+			host := List[goos][goarch]
+			if host != nil && host.PtrSize > target.PtrSize {
+				// Consider the 32 bit pair target instead.
+				host = List[goos][host.Target32]
+			}
+			if host != nil {
 				target.CCompiler = host.CCompiler
 				target.CPP = host.CPP
+				addCFlags := append([]string{}, host.CFlags...)
 				if goos == FreeBSD {
 					// For some configurations -no-pie is passed to the compiler,
 					// which is not used by clang.
 					// Ensure clang does not complain about it.
-					target.CFlags = append(target.CFlags, "-Wno-unused-command-line-argument")
+					addCFlags = append(addCFlags, "-Wno-unused-command-line-argument")
 					// When building executor for the test OS, clang needs
 					// to link against the libc++ library.
-					target.CFlags = append(target.CFlags, "-lc++")
+					addCFlags = append(addCFlags, "-lc++")
 				}
+				target.CFlags = fixMergedCFlags(append(append([]string{}, target.CFlags...), addCFlags...))
```

Comment:
Why can't we append to target.CFlags as currently and filter target.CFlags?

Reply:
Or since the latter flags override the earlier flags, can't we just do:
```
target.CFlags = append(append([]string{}, host.CFlags...) target.CFlags...)
```
and don't do fixMergedCFlags?